objc/src/ort_session.mm

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import "onnxruntime/ort_session.h"
+#import "onnxruntime/ort_env.h"
+#import "onnxruntime/ort_value.h"
+#import "src/error_utils.h"
+#import "src/ort_env_internal.h"
+#import "src/ort_value_internal.h"
+
+#include <vector>
+
+#include "core/common/optional.h"
+#include "core/session/onnxruntime_cxx_api.h"
+
+@implementation ORTSession {
+  onnxruntime::optional<Ort::Session> _session;
```

Comment:
>onnxruntime::optional [](http://example.com/codeflow?start=2&length=21)

Is there a particular reason to use optional?
Can we use std::unique_ptr instead, such that we can have single dependancy (onnxruntime_cxx_api.h)?

Reply:
the reason for using optional is to avoid another dynamic allocation. once we move to C++17 we can use std::optional

---
In reply to: [616178018](https://github.com/microsoft/onnxruntime/pull/7366#discussion_r616178018) [](ancestors = 616178018)

Reply:
Can we force CXX_STANDARD 17 for onnxruntime_objc directly?

Reply:
hm, let me try that :)

---
In reply to: [616234266](https://github.com/microsoft/onnxruntime/pull/7366#discussion_r616234266) [](ancestors = 616234266)